application/backend/src/main/java/pl/ds/bulma/components/helpers/ColorService.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2022 Dynamic Solutions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pl.ds.bulma.components.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+
+public class ColorService {
+  private final List<String> colorBw;
+  private final List<String> colorGrey;
+  private final List<String> colorRest;
+  private final String color;
+  private final String shadeBw;
+  private final String shadeGrey;
+  private final String shadeRest;
+
+  public ColorService(Resource resource, String resourcePath, String color,
+                      String shadeBw, String shadeGrey, String shadeRest) {
+    List<String> colorsList = getColorsListFromResource(resource, resourcePath);
+
+    this.colorBw = getElementsFromListByPrefix(colorsList, "bw_");
```

Comment:
I'm wondering if we could read the prefix (shade_) and map colors by it. Then we could find shades by this prefix. It should be more elastic. I mean, in the project we could overwrite the whole colors and shades with something totally different.

Reply:
Accepted for now. We can refactor it later.